gm4_mountaineering/data/gm4_mountaineering/functions/player_main.mcfunction

```diff
@@ -0,0 +1,22 @@
+# Submain function to limit `as @a at @s` in main
+# @s = player not in spectator
+# at @s
+# run from main
+
+# Run ski-related functions
+execute at @s if entity @s[predicate=gm4_mountaineering:wearing_skis] run function gm4_mountaineering:ski_effects/skis_equipped
```

Comment:
I've gone for `at @s[predicate=gm4_mountaineering:wearing_skis]` now, so there aren't any options for putting `at @s` in any particular order.

I take it this is the best way? Or did you mean to put `at @s` and then the `if entity` subcommand? (I assumed not the latter, since we want to keep subcommands down)